main.go

```diff
@@ -317,13 +260,37 @@ func failf(format string, v ...interface{}) {
 	os.Exit(1)
 }
 
+func validate(cfg *configs) error {
+	if cfg.APKPath != "" {
+		log.Warnf("step input 'APK file path' (apk_path) is deprecated and will be removed on 20 August 2019, use 'APK or App Bundle file path' (android_app) instead!")
+		cfg.BuildArtifactPath = cfg.APKPath
+	}
+
+	buildArtifactPaths := parseAppList(cfg.BuildArtifactPath)
+	for _, buildArtifactPath := range buildArtifactPaths {
+		if exist, err := pathutil.IsPathExists(buildArtifactPath); err != nil {
+			return fmt.Errorf("failed to check if BuildArtifactPath exist at: %s, error: %s", buildArtifactPath, err)
+		} else if !exist {
+			return fmt.Errorf("BuildArtifactPath not exist at: %s", buildArtifactPath)
+		}
+	}
+	return nil
+}
+
 // -----------------------
 // --- Main
 // -----------------------
 func main() {
-	configs := createConfigsModelFromEnvs()
-	configs.print()
-	if err := configs.validate(); err != nil {
+	var cfg configs
+	if err := stepconf.Parse(&cfg); err != nil {
+		failf("Issue with input: %s", err)
+	}
+
+	stepconf.Print(cfg)
+	log.SetEnableDebugLog(cfg.VerboseLog)
+	fmt.Println()
+
+	if err := validate(&cfg); err != nil {
```

Comment:
the name `validate` doesn't cover the fact that the method modifies any of the model fields. so it is not only a validation

Reply:
Fixed